### PR TITLE
[TEST] Reload secure settings transport IT

### PR DESCRIPTION
**Pull request by albertzaharovits**
Adds Integration Tests for the `TransportNodesReloadSecureSettingsAction` . Specifically, it tests:
* SecureSettings are retrievable when plugins are notified
* missing/invalid keystore is handled gracefully (no spurious `reload` calls)
* reloadable plugins should fail independently (a `throw` during a plugin reload on any node will not interrupt the `reload` flow in any way - it will just return the exception in the response)

Previously there were tests only for the individual plugins implementing the reload handler, no tests for the "broadcast password - decrypt keystore - notify plugins - close keystore" logic.

One obstacle I struggled with was generating keystores programatically. They are test resources in the present implementation. The problem is that saving a generated keystores requires the Java SM permissions to change file permissions. This is not granted for the server codebase. In the real world the keystore is generated before the Security Manager is installed.

Relates: https://github.com/elastic/elasticsearch/issues/29135
CC @elastic/es-security

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java
Comment (albertzaharovits) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java]: 1/2 non test change

File: server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java
Comment (albertzaharovits) [inline server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java]: 2/2 non test change
This password is no longer internal API, there are code paths where the user supplies its value.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java
Comment (albertzaharovits) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java]: 1/2 non test change

File: server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java
Comment (jaymode) [inline server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java]: Collections.singletonList

File: server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java
Comment (jaymode) [inline server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java]: can we rename this to `shouldThrow`? Then just have a setter for the method? Also, let's make this non-volatile and the methods synchronized since we technically do a CAS in reload

**Review by jaymode — CHANGES_REQUESTED** [review]
I left some comments. Regarding not being able to write a keystore in tests, maybe someone from @elastic/es-core-infra could provide some thoughts about whether there is a better way than checking the files in as a resource.

File: server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java
Comment (albertzaharovits) [reply server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java]: Yes! I was under the impression that the order here is not relevant when iterating oveer Plugin instances. But it is, and this is a good thing!

File: server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java
Comment (albertzaharovits) [reply server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java]: Yes, I've been lax here.

**Comment by albertzaharovits** [comment]
Thank you @jaymode .
I have addressed your comments. I will wait a little for suggestions, but probably start pulling someone's coat over slack.

**Review by jaymode — APPROVED** [review]
LGTM pending discussion with the core-infra team regarding the updating of the keystore during the test

File: server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java
Comment (jaymode) [inline server/src/test/java/org/elasticsearch/action/admin/ReloadSecureSettingsIT.java]: you need to use `Collections.shuffle(plugins, random())` to fix the build failure

**Review by jaymode — APPROVED** [review]
Left one comment. Otherwise LGTM as long as CI is happy.

**Comment by albertzaharovits** [comment]
Appreciate the prompt review! Thank you, I will make sure CI runs several time, before merging the feature branch!

**Pull request merged**
